core/src/main/java/io/varhttp/ControllerExecution.java

```diff
@@ -118,19 +81,28 @@ public Method getMethod() {
 	}
 
 	private static class VarFilterChain implements FilterChain {
-		private final Filter current;
+		private ControllerContext context;
+		private final Object current;
 		private FilterChain chain = null;
 
-		public VarFilterChain(Filter current, Iterator<Filter> iterator) {
+		public VarFilterChain(ControllerContext context, Object current, Iterator<Object> iterator) {
+			this.context = context;
 			this.current = current;
 			if (iterator.hasNext()) {
-				chain = new VarFilterChain(iterator.next(), iterator);
+				chain = new VarFilterChain(context, iterator.next(), iterator);
 			}
 		}
 
 		@Override
 		public void doFilter(ServletRequest request, ServletResponse response) throws IOException, ServletException {
-			current.doFilter(request, response, chain);
+			if (current instanceof Filter) {
+				((Filter) current).doFilter(request, response, chain);
+			} else if(current instanceof VarFilterExecution) {
+				((VarFilterExecution) current).doFilter(context);
+				chain.doFilter(request, response);
```

Comment:
not sure if this logic should be here or inside the doFilter. Nevertheless, it should be done by default.